packaging/build_rpm.py

```diff
@@ -336,22 +338,46 @@ def install_mock():
 
 
 def check_mock_config():
-    # allow network access during the build
-    # (we have to download packages from pypi)
-    mock_config = check_output(
-        MOCK + ['--debug-config',
-        ]).splitlines()
-
-    for line in mock_config:
-        if all(s in line for s in ('rpmbuild_networking', 'True')):
-            break
-    else:
-        logger.info('configuring rpmbuild_networking')
-        check_call(
-                '''echo "config_opts['rpmbuild_networking'] = True" '''
-                '| sudo -n tee -a /etc/mock/site-defaults.cfg',
-                shell=True,
-                )
+    """
+    allow network access during the build and set the %dist macro
+    (we have to download packages from pypi)
+    """
+    mock_config = check_output(MOCK + ['--debug-config', '-qq'])
+
+    config_opts = {}
+    config = {'config_opts': config_opts}
+    exec mock_config in config, config
+
+    options = [
+        "config_opts['rpmbuild_networking'] = True",
+        "config_opts['macros']['%dist'] = '.el7'",
+    ]
+    for option in reversed(options):
+        lhs, eq, rhs = option.partition(' = ')
+        if not eq:
+            raise ValueError('malformed option', option)
+
+        try:
+            exists = eval('{} == {}'.format(lhs, rhs), config, config)
```

Comment:
do we actually need to check? i think we can simply append our config values at the end of the config file, not conditionally, without any checking. I do realize it's not as nice, but in the realistic case, it should be good enough? and it's much simpler

Reply:
That means on the mock builder this file will continuously grow, until eventually in a year's time the disk is full and it breaks